### PR TITLE
Add logging to verifySubmission method

### PR DESCRIPTION
**Pull request by HarryBosworth**
Added logging to the `verifySubmission` method by testing the result and logging on failure.
Adds log message and appropriate details to the log depending on which condition failed.

BI-6635

### CODE REVIEW

**Comment by PhilipGlover** [comment]
when I do a make clean build I get:
```[ERROR] Errors:
[ERROR]   PaymentControllerImplTest>BaseControllerImplTest.verifySubmissionLogsOnFailure:172 » NullPointer
[ERROR]   PaymentControllerImplTest>BaseControllerImplTest.verifySubmissionLogsOnFailure:172 » NullPointer
[ERROR]   PaymentControllerImplTest>BaseControllerImplTest.verifySubmissionLogsOnFailure:172 » NullPointer
[INFO]
[ERROR] Tests run: 337, Failures: 0, Errors: 3, Skipped: 0```

**Comment by HarryBosworth** [comment]
I've removed the over-ridden logger in PaymentController and done a full build to make sure all tests pass

File: src/test/java/uk/gov/companieshouse/efs/web/controller/BaseControllerImplTest.java [hunk 1]
```diff
@@ -130,4 +158,63 @@ protected SubmissionApi createSubmission(final SubmissionStatus submitted) {
 
         return submission;
     }
+
+    @ParameterizedTest
+    @MethodSource("verifySubmissionTestCases")
+    void verifySubmissionLogsOnFailure(String reqSubmissionID, String sessionSubmissionID,
+                                       String requestUserEmail, String sessionUserEmail,
+                                       boolean isSamForm, boolean isSameUser) {
```
Comment (PhilipGlover) [inline src/test/java/uk/gov/companieshouse/efs/web/controller/BaseControllerImplTest.java:166]: typo 'isSamForm'

**Review by PhilipGlover — APPROVED** [review]
nicely done and like the parameterized test

**Pull request merged**